### PR TITLE
Fix DecimalFormat when BigDecimal value is 0

### PR DESCRIPTION
**Pull request by cathyzhyi**
The code deciding whether to print out the decimal separator
doesn't work with 0. This changeset fixes this problem so that
the result string won't contain a redundant decimal separator.

Signed-off-by: Yi Zhang <yizhang@ca.ibm.com>

### CODE REVIEW

File: jcl/src/java.base/share/classes/com/ibm/jit/DecimalFormatHelper.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 import java.lang.String;
 import java.lang.RuntimeException;
 /*******************************************************************************
- * Copyright (c) 2011, 2011 IBM Corp. and others
+ * Copyright (c) 2018, 2018 IBM Corp. and others
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/com/ibm/jit/DecimalFormatHelper.java]: The start date should remain 2011

Comment (cathyzhyi) [reply jcl/src/java.base/share/classes/com/ibm/jit/DecimalFormatHelper.java]: changed. thanks!

**Comment by pshipton** [comment]
jenkins test sanity

**Comment by pshipton** [comment]
Is this fix a bug in active code, or preparation for making the code active?

**Comment by cathyzhyi** [comment]
This is a bug in active code.

**Comment by pshipton** [comment]
Is there a test that can be updated to cover this case?

**Comment by pshipton** [comment]
Can the testcase be added to this Pull Request? Then the Pull Request testing can run the test and ensure it is passing with this fix.

**Comment by pshipton** [comment]
Note the copyright check failed

File: test/JIT_Test/src/jit/test/tr/stringPeephole/BigDecimalToStringTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.math.BigDecimal;
+import java.text.NumberFormat;
+import java.text.DecimalFormat;
+
+@Test(groups = { "level.sanity","component.jit" })
+
+public class BigDecimalToStringTest{
+
+      private static Logger logger = Logger.getLogger(BigDecimalToStringTest.class);
```
Comment (smlambert) [inline test/JIT_Test/src/jit/test/tr/stringPeephole/BigDecimalToStringTest.java]: You do not use logger anywhere in the test code.  Not sure if there is any need for test output, as your Assert will print to console, but perhaps you may want some logger.debug statements (which don't print to console in the regular runs, but tests can be rerun in that logging level if you are troubleshooting stuff.  

I am fine to leave it here, if there are plans to expand this test class and log output in near future.

File: test/JIT_Test/src/jit/test/tr/stringPeephole/BigDecimalToStringTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+         // Run tests enough time to force compilation
+         for (int i=0; i< numIters; i++) 
+            testBigDecimal0ToString();
+      }
+
+      public void testBigDecimal0ToString() {
+         NumberFormat format = new DecimalFormat("###,###,##0");
+         BigDecimal bigDecimalValue = new BigDecimal(new Double(0.000));
+         String s = bigDecimalToString(format, bigDecimalValue);
+         AssertJUnit.assertEquals(s, "0");
```
Comment (smlambert) [inline test/JIT_Test/src/jit/test/tr/stringPeephole/BigDecimalToStringTest.java]: You don't actually need to use this AssertJUnit class, as this is a brand new test (I see we have to update the TestExample class).  You can use org.testng.Assert class instead, the AssertJUnit class is used for when we are converting old JUnit tests to testNG and we don't want to have to change the order of the assert parameters en masse.  Condition and string are reversed from the JUnit format.

File: test/JIT_Test/src/jit/test/tr/stringPeephole/BigDecimalToStringTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2001, 2018 IBM Corp. and others
```
Comment (smlambert) [inline test/JIT_Test/src/jit/test/tr/stringPeephole/BigDecimalToStringTest.java]: if this is a brand new test, copyright might not start at 2001

File: test/JIT_Test/playlist.xml [hunk 5]
```diff
@@ -217,6 +217,7 @@
 	mathSimplifyTest,\
 	NewInstanceTest,\
 	signExtensionATest,\
+	StringPeepholeTest,\
```
Comment (smlambert) [inline test/JIT_Test/playlist.xml]: this is added to a suite that is run with "-Xjit:noJitUntilMain -Xdump:java" as the commandline options, is that sufficient for your test?  You mentioned a few others in our conversation "-Xjit:{*theMethodWithCodePatternfortesting*}(count=0,optlevel=hot),dontinline={*theMethodWithCodePatternfortesting*}"

**Comment by smlambert** [comment]
Jenkins test sanity zLinux

**Comment by smlambert** [comment]
Test changes look good.

**Comment by vijaysun-omr** [comment]
Change looks reasonable to me. Merging.

**Pull request merged**